crates/graphql-schema-diff/src/path/parse.rs

```diff
@@ -0,0 +1,173 @@
+use super::*;
+
+impl<'a> Path<'a> {
+    pub fn parse(s: &'a str) -> ParseResult<Path<'a>> {
+        let mut segments = s.split('.');
+
+        let first_segment = segments.next().filter(|s| !s.is_empty()).ok_or(ParseError)?;
+
+        // Is it a schema definition or extension?
+        if let Some(suffix) = first_segment.strip_prefix(":schema") {
+            if segments.next().is_some() {
+                return Err(ParseError);
+            }
```

Comment:
Take it we don't care about directives on schema definitions?

Reply:
It's not implemented yet ([issue](https://linear.app/grafbase/issue/GB-7389/graphql-schema-diff-implement-directive-addition-and-removal-as-well))